src/span.ts

```diff
@@ -1,12 +1,14 @@
 import * as noop from './noop';
+import SpanContext from './span_context';
+import Tracer from './tracer';
 
 /**
  * Span represents a logical unit of work as part of a broader Trace. Examples
  * of span might include remote procedure calls or a in-process function calls
  * to sub-components. A Trace has a single, top-level "root" Span that in turn
  * may have zero or more child Spans, which in turn may have children.
  */
-export default class Span {
+export class Span {
```

Comment:
By coupling I mean the workarounds with initialization for Noop due to circular dependencies. If `opentracing.Tracer` itself is meant to be a concrete implementation, then it has to be a noop implementation, so there's no reason to have Noop as something separate, it just complicates the library.

I am not too concerned with backwards compatibility given v1.0 label (the external API is still the same). But we can keep some of the shared functionality like normalizing references in StartSpan options. I'd just prefer to see a lot fewer underscore methods, especially when the delegation is completly pass through, as pretty much in all Span methods.